programs/fileio.c

```diff
@@ -1453,6 +1453,9 @@ int FIO_compressMultipleFilenames(FIO_prefs_t* const prefs,
     }   }
 
     FIO_freeCResources(ress);
+    if (dstFileNamesTable)
+        UTIL_freeDestinationFilenameTable(dstFileNamesTable, nbFiles);
```

Comment:
Programming patterns :
for better code maintenance, it's recommended
to free an object in the same code block as the one where it's created.

```C
{
    TYPE* const object = createObject();
    (...)
    freeObject(object);
}
```

It seems this is possible here too.